OneCommunityGlobal/HighestGoodNetworkApp
# Tim: Update & fix Material List, Material Update pages

# Description
Updated Material List and Update pages to work with updated data model. Fixed data structure set by reducer to prevent build time page crashes.

## Related PRS (if any):
[Backed PR 656](https://github.com/OneCommunityGlobal/HGNRest/pull/656) must be running

## Main changes explained:
- Material pages no longer crash
- Code refactored
- Files renamed to remove confusing pluralization

## How to test:
1. `git checkout Tim_update_mat_pages`
2. `nvm i`, then `npm i`, then `npm run start:local`
3. Clear site data/cache
4. Log in as volunteer
5. Navigate to "/bmdashboard" and confirm credentials
6. Navigate to Material List page: "/bmdashboard/materials"
7. Check that the page does not crash. Check Updates and Purchases modals for accurate display of records.
   - If the page does crash, try [clearing your browser's cache and cookies](https://support.google.com/accounts/answer/32050)
9. Click the pencil icon to add an update record for an item. Check that data submits correctly and the page is updated to reflect new values.
10. Navigate to Material Updates page: "/bmdashboard/materials/update"
11. Check that the page does not crash. Test the form.
   - If the page does crash, try [clearing your browser's cache and cookies](https://support.google.com/accounts/answer/32050)
